File: php/spec/ipv6.spec.php
<?php declare(strict_types=1);

use Enotype\Loaders;

describe('ipv6', function() {
  describe('with \'::1\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::ipv6('::1'))->toEqual('::1');
    });
  });
  
  describe('with \'2001:db8::\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::ipv6('2001:db8::'))->toEqual('2001:db8::');
    });
  });
  
  describe('with \'0:0:0:0:0:0:0:0\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::ipv6('0:0:0:0:0:0:0:0'))->toEqual('0:0:0:0:0:0:0:0');
    });
  });
  
  describe('with \':::1\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::ipv6(':::1'); })->toThrow();
    });
  });
  
  describe('with \'::0::1\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::ipv6('::0::1'); })->toThrow();
    });
  });
  
  describe('with \'localhost\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::ipv6('localhost'); })->toThrow();
    });
  });
  
  describe('with \'0.0.0.0\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::ipv6('0.0.0.0'); })->toThrow();
    });
  });
});

File: ruby/spec/ipv6.spec.rb
require 'enotype'

describe 'ipv6' do
  context 'with \'::1\'' do
    it 'produces the expected result' do
      expect(Enotype::ipv6('::1')).to eq('::1')
    end
  end
  
  context 'with \'2001:db8::\'' do
    it 'produces the expected result' do
      expect(Enotype::ipv6('2001:db8::')).to eq('2001:db8::')
    end
  end
  
  context 'with \'0:0:0:0:0:0:0:0\'' do
    it 'produces the expected result' do
      expect(Enotype::ipv6('0:0:0:0:0:0:0:0')).to eq('0:0:0:0:0:0:0:0')
    end
  end
  
  context 'with \':::1\'' do
    it 'raises an error' do
      expect { Enotype::ipv6(':::1') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'::0::1\'' do
    it 'raises an error' do
      expect { Enotype::ipv6('::0::1') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'localhost\'' do
    it 'raises an error' do
      expect { Enotype::ipv6('localhost') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'0.0.0.0\'' do
    it 'raises an error' do
      expect { Enotype::ipv6('0.0.0.0') }.to raise_error(RuntimeError)
    end
  end
end